topic-offset-service/src/main/java/uk/gov/gchq/palisade/service/topicoffset/model/TopicOffsetRequest.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.topicoffset.model;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import uk.gov.gchq.palisade.Context;
+import uk.gov.gchq.palisade.Generated;
+import uk.gov.gchq.palisade.resource.LeafResource;
+import uk.gov.gchq.palisade.resource.Resource;
+
+import java.util.Optional;
+
+/**
+ * The TopicOffsetRequest is the input for topic-offset-service.  This is used to optimise the later retrieval of the
+ * data for the client. TopicOffsetResponse is the output for this service.
+ * Note there are three classes that effectively represent the same data but represent a different stage of the process.
+ * uk.gov.gchq.palisade.service.attributemask.message.TopicOffsetResponse is the output from the policy-service.
+ * uk.gov.gchq.palisade.service.topicoffset.message.TopicOffsetRequest is the input for this the topic-offset-service.
+ * uk.gov.gchq.palisade.service.filteredresource.message.FilteredResourceRequest is the input for this the
+ * filtered-resource-service.
```

Comment:
This doesn't seem right.
The TopicOffsetResponse is the output of the topic-offset-service, not the Policy-Service.
The input for the filtered-resource-service should be the `TopicOffsetMessage` class?
```suggestion
 * Note there are three classes that effectively represent the same data but represent a different stage of the process.
 * uk.gov.gchq.palisade.service.topicoffset.message.TopicOffsetRequest is the input for this the topic-offset-service.
  * uk.gov.gchq.palisade.service.attributemask.message.TopicOffsetResponse is the output from the topic-offset-service.
 * uk.gov.gchq.palisade.service.filteredresource.message.TopicOffsetMessage is the input for the
 * filtered-resource-service.
```